go/lru/disk_lru.go

```diff
@@ -384,3 +386,67 @@ func (d *DiskLRU) Size(ctx context.Context, lctx libkb.LRUContext) (int, error)
 	}
 	return index.Size(), nil
 }
+
+func (d *DiskLRU) allValuesLocked(ctx context.Context, lctx libkb.LRUContext) (entries []DiskLRUEntry, err error) {
+	var index *diskLRUIndex
+	defer func() {
+		// Commit the index
+		if err == nil && index != nil && index.IsDirty() {
+			d.writeIndex(ctx, lctx, index, false)
+		}
+	}()
+
+	// Grab entry index
+	index, err = d.readIndex(ctx, lctx)
+	if err != nil {
+		return nil, err
+	}
+	for key := range index.entryKeyMap {
+		if found, res, err := d.readEntry(ctx, lctx, key); err != nil {
+			return nil, err
+		} else if !found {
+			index.Remove(key)
+		} else {
+			entries = append(entries, res)
+		}
+	}
+	return entries, nil
+}
+
+func (d *DiskLRU) Clean(ctx context.Context, lctx libkb.LRUContext, cacheDir string) (err error) {
+	d.Lock()
+	defer d.Unlock()
+
+	// reverse map of filepaths to lru keys
+	cacheRevMap := map[string]string{}
+	allVals, err := d.allValuesLocked(ctx, lctx)
+	if err != nil {
+		return err
+	}
+	for _, entry := range allVals {
+		path, ok := entry.Value.(string)
+		if !ok {
+			continue
+		}
+		cacheRevMap[path] = entry.Key
+	}
+
+	if err := os.MkdirAll(cacheDir, os.ModePerm); err != nil {
```

Comment:
What is this for?

Reply:
santiy check to make sure the dir exists but i guess not necessary since glob ignores missing dirs https://golang.org/pkg/path/filepath/#Glob

Reply:
Yeah it was a little confusing to me, since if it isn't there, then presumably there is nothing to clean.